src/Xamarin.Android.Build.Tasks/Tasks/GenerateResourceDesigner.cs

```diff
@@ -90,7 +90,12 @@ public override bool RunTask ()
 					throw new ArgumentNullException (nameof (JavaResgenInputFile));
 				}
 				var parser = new JavaResourceParser () { Log = Log };
-				resources = parser.Parse (JavaResgenInputFile, IsApplication, resource_fixup);
+				var parsedResources = parser.Parse (JavaResgenInputFile, IsApplication, resource_fixup);
+				if (parsedResources == null) {
+					Log.LogError ("Failed to parse Java resource file: " + JavaResgenInputFile);
+					return false;
+				}
+				resources = parsedResources;
```

Comment:
Can we simplify this to:
```suggestion
				resources = parser.Parse (JavaResgenInputFile, IsApplication, resource_fixup);
				if (resources == null) {
					throw new ArgumentNullException (nameof (resources));
				}
```